src/utils/use-on-pause.ts

```diff
@@ -0,0 +1,18 @@
+import {useRef, useEffect} from 'react';
+import {AppState} from 'react-native';
+
+export function useOnPause(runOnPauseFunc: () => void) {
+  const appState = useRef(AppState.currentState);
+
+  useEffect(() => {
+    const subscription = AppState.addEventListener('change', (nextAppState) => {
+      if (appState.current.match(/active/) && nextAppState === 'background') {
```

Comment:
What if the state change goes 'active' -> 'inactive' -> 'background'?

Reply:
`inactive` will also be matched by `.match(/active/)`, but I guess it is better to be more explicit about it :+1: